applications/segment_everything/Dockerfile

```diff
@@ -0,0 +1,57 @@
+# syntax=docker/dockerfile:1
+
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ARG BASE_IMAGE
+ARG GPU_TYPE
+
+FROM ${BASE_IMAGE} as base
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+# --------------------------------------------------------------------------
+#
+# Install required apt packages and python modules
+#
+RUN apt-get update \
+    && apt-get install -y libgl1
+
+RUN pip3 install \
+    onnx\
+    torch\
+    torchvision\
+    torchaudio\
+    matplotlib
+# clone the repo of SAM
+RUN git clone https://github.com/facebookresearch/segment-anything.git \
+&& cd segment-anything \
+&& pip install -e .
```

Comment:
We should use the requirements.txt file (easier to maitain)
See: https://github.com/nvidia-holoscan/holohub/blob/9c7d50c14faf2d42044132633d217ac11024b49f/Dockerfile#L70C2-L71C54

Reply:
I removed the entire Dockerfile. It is not needed anymore, since the segment_anything repo is not used here, but in the seperate repo, to create the engine files. 
The default dockerfile from holohub is sufficient then.